homeassistant/components/caldav/calendar.py

```diff
@@ -51,6 +52,9 @@
                         vol.Required(CONF_CALENDAR): cv.string,
                         vol.Required(CONF_NAME): cv.string,
                         vol.Required(CONF_SEARCH): cv.string,
+                        vol.Optional(
+                            CONF_DEFAULTOFFSET, default="00:00"
+                        ): cv.time_period_str,
```

Comment:
We no longer accept YAML configuration for integrations that connect to devices or services.

Instead, a configuration flow via the UI should be used.

For more information about this decision see:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>
- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

To get started on the integration configuration, see our developer documentation:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

Reply:
So then it would be better to convert the caldav integration to the new configuration flow and then add this new feature?

Reply:
Yes, please do the refactor first. Then do another PR with the new feature.